platform/marvell-arm64/sonic-platform-modules-supermicro/sse-g3748/sonic_platform/sfp.py

```diff
@@ -0,0 +1,1028 @@
+#############################################################################
+# SSE-G3748
+#
+# There are 2 IO Expander connect to SFP transceiver status/control
+#---------------------------------------------------------------------------
+# PCAL6524 -- [I2C 0x22]    24 bits I/O for SFP status and TX_DIS
+#   Port 0 --    7      6      5      4        3      2      1      0
+#            TX_FAULT TX_DIS  LOS  PRESENT TX_FAULT TX_DIS  LOS  PRESENT
+#              SFP1    SFP1   SFP1  SFP1     SFP0    SFP0   SFP0  SFP0 
+#
+#   Port 1 --    7      6      5      4        3      2      1      0
+#            TX_FAULT TX_DIS  LOS  PRESENT TX_FAULT TX_DIS  LOS  PRESENT
+#              SFP3    SFP3   SFP3  SFP3     SFP2    SFP2   SFP2  SFP2
+#
+#   Port 1 --    7      6      5      4        3      2      1      0
+#            TX_FAULT TX_DIS  LOS  PRESENT TX_FAULT TX_DIS  LOS  PRESENT
+#              SFP5    SFP5   SFP5  SFP5     SFP4    SFP4   SFP4  SFP4
+# [TX_DIS bit is output, all other bits are input]
+#---------------------------------------------------------------------------
+# PCAL6416A -- [I2C 0x20]  16 bits I/) Expander  SFP RS0 and RS1
+#    Port 0 --   7    6    5    4        3    2    1    0
+#               RS0  RS1  RS0  RS1      RS0  RS1  RS0  RS1
+#                  SF2       SF3          SF0       SF1
+#
+#    Port 0 --   7    6    5    4        3    2    1    0
+#               N/A  N/A  N/A  N/A      RS0  RS1  RS0  RS1
+#                                         SF4       SF5
+# [ All bits are output]
+#
+#############################################################################
+
+import os
+import sys
+
+try:
+    from sonic_platform_base.sfp_base import SfpBase
+    from sonic_platform_base.sonic_sfp.sff8472 import sff8472InterfaceId
+    from sonic_platform_base.sonic_sfp.sff8472 import sff8472Dom
+    from sonic_platform_base.sonic_sfp.sfputilhelper import SfpUtilHelper
+    from sonic_py_common import logger
+except ImportError as e:
```

Comment:
@SelvaSupermicro Please use SFP refactor code. This is the HLD [https://github.com/sonic-net/SONiC/blob/master/doc/sfp-refactor/sfp-refactor.md](https://github.com/sonic-net/SONiC/blob/master/doc/sfp-refactor/sfp-refactor.md)

Reply:
@prgeor As per my analysis SFP refactor started from 202111. Please not that this commit is in 202106.
Please educate me if I am wrong..

Reply:
@SelvaSupermicro SFP refactor is supported from 202205 onwards.

Reply:
@prgeor Sorry for the confusion, I actually meant that I am pushing this commit in 202106 which does not support SFP refactor. As you have mentioned SFP refactor is applicable only from 202205.